### DISCUSSION TITLE
launch configuration for IntelliJ IDEA allowing developing with debug feature

### DISCUSSION THREAD

**Discussion by Nevysha** [anapnoe/stable-diffusion-webui-ux#104] [Show and tell]
Hello !

I don't know where to post it, sorry if this is not the right place.

As a developer I like to deep dive inside the code to learn and understand how things work. I'm interrested in contributing on the UI/UX part of this repo (not on the core code, since I read that it will be merged with [Vlad's version](https://github.com/vladmandic/automatic)

I'm not sure everyone is confortable with IntelliJ launch config so here is one that allow you to launch the webui directly from IntelliJ, and using the debug tool to run step by step code, seeing the variable, etc.

The important part is setting the "envs" element with env variable to be able to bypass the bat/sh script usually used to launch the webui.

Hope it may help someone,
Regards

```
<component name="ProjectRunConfigurationManager">
  <configuration default="false" name="launch" type="PythonConfigurationType" factoryName="Python">
    <module name="stable-diffusion-webui-ux" />
    <option name="INTERPRETER_OPTIONS" value="" />
    <option name="PARENT_ENVS" value="true" />
    <envs>
      <env name="PYTHONUNBUFFERED" value="1" />
      <env name="COMMANDLINE_ARGS" value="--xformers --api --opt-split-attention-v1 --use-cpu=interrogate --use-cpu=esrgan" />
    </envs>
    <option name="SDK_HOME" value="$PROJECT_DIR$/venv/Scripts/python.exe" />
    <option name="SDK_NAME" value="Python 3.10 stable-diffusion-webui-ux (venv)" />
    <option name="WORKING_DIRECTORY" value="$PROJECT_DIR$" />
    <option name="IS_MODULE_SDK" value="false" />
    <option name="ADD_CONTENT_ROOTS" value="true" />
    <option name="ADD_SOURCE_ROOTS" value="true" />
    <EXTENSION ID="PythonCoverageRunConfigurationExtension" runner="coverage.py" />
    <option name="SCRIPT_NAME" value="$PROJECT_DIR$/launch.py" />
    <option name="PARAMETERS" value="" />
    <option name="SHOW_COMMAND_LINE" value="false" />
    <option name="EMULATE_TERMINAL" value="false" />
    <option name="MODULE_MODE" value="false" />
    <option name="REDIRECT_INPUT" value="false" />
    <option name="INPUT_FILE" value="" />
    <method v="2" />
  </configuration>
</component>
```

**Comment by anapnoe** [comment]
we will keep our repositories and we hope to create an organization with many admins/maintainers
you are welcome to join our dev discord server https://discord.gg/R46Xmx8B
